python/tank/bootstrap/configuration.py

```diff
@@ -162,7 +177,17 @@ def _set_authenticated_user(self, user):
 
             log.debug("%r will be used.", authenticated_user)
 
+            # If we're logging in with the human user, try to reinstantiate it with the new core API.
+            if authenticated_user == user:
```

Comment:
let's discuss this tomorrow - my worry is that we are touching the user object, comparing it etc. It doesn't feel like an emergency fallback - we are making assumptions about this potential unknown black box. Perhaps a minor change, but could we do something along these lines?

```python
if hasattr(api, "set_authenticated_user"):
    # modern world of toolkit yay
    authenticated_user = None
    default_user = do_logic_around_default_users_script_users_etc()
    if default_user:
         # for reasons logged and pointed out in comments, we are ignoring whatever the 
         # old core passed to us and we are running with our own credentials setup
         authenticated_user = default_user
    else:
         # feed this new core the user info we got from the old core
         authenticated_user = try_to_deserialize_user(serialized_user)
         if authenticated_user is None:
              # for some reason (missing deserialize API) we do not understand serialized_user
              # as a last fallback, take the user object from the old core we were swapping from 
              # and use that as the authenticated user in the new core that we are swapping to
              logger.warning("uh-oh - passing the user object from one core to another!")
              api.set_authenticated_user(user)
```

Maybe you think i am being nuts, but i just think it would be cool to know that that `user` parameter (from the old swapped-from core) is only being touched and accessed as a last resort. We are not even doing equality operations on it and stuff like that, because we don't know what it is.

Hope the pseudo code makes sense. Let's chat about it! I know it's minor but i always worry about the core swap and about future maintenance of the core swap and i think the clearer we can make it, the better. That's why i was asking about whether we could just push the serialized strings across initially. (but i totally get why that can be super tricky)